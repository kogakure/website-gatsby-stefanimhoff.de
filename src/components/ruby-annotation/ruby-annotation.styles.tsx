import styled from 'styled-components';

import { fontSize } from '../../layout/tokens';

const Ruby = styled.ruby`
  font-size: ${fontSize[3]};
`;

export const Styled = {
  Ruby,
};
