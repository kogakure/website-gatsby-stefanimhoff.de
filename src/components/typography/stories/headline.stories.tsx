import React from 'react';

import { Headline } from '..';

export default {
  title: 'Typography/Headline',
};

export const Default = () => <Headline>Headline</Headline>;
