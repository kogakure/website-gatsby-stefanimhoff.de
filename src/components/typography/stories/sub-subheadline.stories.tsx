import React from 'react';

import { SubSubheadline } from '..';

export default {
  title: 'Typography/SubSubheadline',
};

export const Default = () => <SubSubheadline>SubSubheadline</SubSubheadline>;

export const NoMargin = () => (
  <SubSubheadline noMargin>SubSubheadline</SubSubheadline>
);
