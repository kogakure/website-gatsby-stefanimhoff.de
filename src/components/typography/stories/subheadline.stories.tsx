import React from 'react';

import { Subheadline } from '..';

export default {
  title: 'Typography/Subheadline',
};

export const Default = () => <Subheadline>Subheadline</Subheadline>;

export const NoMargin = () => <Subheadline noMargin>Subheadline</Subheadline>;
