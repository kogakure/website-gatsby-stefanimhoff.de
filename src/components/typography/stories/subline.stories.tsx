import React from 'react';

import { Subline } from '..';

export default {
  title: 'Typography/Subline',
};

export const Default = () => <Subline>Subline</Subline>;
