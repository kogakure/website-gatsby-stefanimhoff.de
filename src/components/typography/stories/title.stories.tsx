import React from 'react';

import { Title } from '..';

export default {
  title: 'Typography/Title',
};

export const Default = () => <Title>Title</Title>;
