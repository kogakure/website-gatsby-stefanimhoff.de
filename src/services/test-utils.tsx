import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
