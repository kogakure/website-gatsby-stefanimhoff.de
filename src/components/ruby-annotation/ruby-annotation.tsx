import * as React from 'react';

import { Styled } from './ruby-annotation.styles';

export type RubyAnnotationProps = {
  base: string;
  text: string;
};

export const RubyAnnotation: React.FC<RubyAnnotationProps> = ({
  base,
  text,
  ...props
}) => (
  <Styled.Ruby {...props}>
    {base}
    <rp>（</rp>
    <rt>{text}</rt>
    <rp>）</rp>
  </Styled.Ruby>
);
