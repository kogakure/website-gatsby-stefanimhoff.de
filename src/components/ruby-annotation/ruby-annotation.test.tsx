import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../services/test-utils';

import { Default } from './ruby-annotation.stories';

describe('RubyAnnotation', () => {
  test('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('is accessible', async () => {
    const { container } = render(<Default />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
