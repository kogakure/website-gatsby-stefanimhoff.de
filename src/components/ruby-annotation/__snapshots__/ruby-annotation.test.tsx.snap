// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RubyAnnotation renders correctly 1`] = `
.c0 {
  font-size: 1.25rem;
}

<p>
  <ruby
    class="c0"
  >
    日
    <rp>
      （
    </rp>
    <rt>
      に
    </rt>
    <rp>
      ）
    </rp>
  </ruby>
  <ruby
    class="c0"
  >
    本
    <rp>
      （
    </rp>
    <rt>
      ほん
    </rt>
    <rp>
      ）
    </rp>
  </ruby>
  <ruby
    class="c0"
  >
    語
    <rp>
      （
    </rp>
    <rt>
      ご
    </rt>
    <rp>
      ）
    </rp>
  </ruby>
  で
  <ruby
    class="c0"
  >
    書
    <rp>
      （
    </rp>
    <rt>
      か
    </rt>
    <rp>
      ）
    </rp>
  </ruby>
  いた
  <ruby
    class="c0"
  >
    作
    <rp>
      （
    </rp>
    <rt>
      さく
    </rt>
    <rp>
      ）
    </rp>
  </ruby>
  <ruby
    class="c0"
  >
    文
    <rp>
      （
    </rp>
    <rt>
      ぶん
    </rt>
    <rp>
      ）
    </rp>
  </ruby>
  です。
</p>
`;
